bundles/io/org.eclipse.smarthome.io.console/src/main/java/org/eclipse/smarthome/io/console/internal/extension/MetadataConsoleCommandExtension.java

```diff
@@ -87,14 +95,29 @@ public void execute(String[] args, Console console) {
         }
     }
 
-    private void listMetadata(Console console, String itemName, String namespace) {
-        MetadataKey key = new MetadataKey(namespace, itemName);
-        Metadata metadata = metadataRegistry.get(key);
-        if (metadata != null) {
-            console.println(metadata.toString());
+    private void listMetadata(Console console, String itemName, String namespace, boolean internal) {
+        if (itemName == null) {
+            metadataRegistry.stream().filter(m -> isInternal(m, internal)).map(Metadata::toString)
+                    .forEach(console::println);
+        } else if (namespace == null) {
+            metadataRegistry.stream().filter(MetadataPredicates.ofItem(itemName)).filter(m -> isInternal(m, internal))
+                    .map(Metadata::toString).forEach(console::println);
+        } else {
+            MetadataKey key = new MetadataKey(namespace, itemName);
+            if (metadataRegistry.isInternalNamespace(namespace) == internal) {
+                Metadata metadata = metadataRegistry.get(key);
+                if (metadata != null) {
+                    console.println(metadata.toString());
+                }
+            }
         }
     }
 
+    @NonNullByDefault
```

Comment:
Missed it on review. We don't use `@NonNullByDefault` on methods but for the whole class.

Please have a look at our "conventions":
https://www.eclipse.org/smarthome/documentation/development/conventions.html#null-annotations

I will fix it in another PR.